### PR TITLE
add hal drivers for rtl87x2g series hal

### PR DESCRIPTION
**Pull request by YuzhuoLiuRTK**
This commit add hal drivers for rtl87x2g series HAL.

HI @ZhiyuanTang17, @RSGraceGuan, Please help review.

### CODE REVIEW

File: bee/drivers/adc/Kconfig [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+menuconfig REALTEK_ADC
```
Comment (ZhiyuanTang17) [inline bee/drivers/adc/Kconfig]: Is “menuconfig REALTEK_ADC_ADC” not right?

File: bee/drivers/adc/src/rtl_common/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+zephyr_library()
+
+zephyr_library_sources_ifdef(CONFIG_REALTEK_ADC_ADC rtl_adc.c)
```
Comment (ZhiyuanTang17) [inline bee/drivers/adc/src/rtl_common/CMakeLists.txt]: CONFIG_REALTEK_ADC_ADC：is it correct?

File: bee/drivers/driver_lib/lib/rtl87x2g/gcc/librtl87x2g_io.a
Comment (ZhiyuanTang17) [inline bee/drivers/driver_lib/lib/rtl87x2g/gcc/librtl87x2g_io.a:1]: what is this lib used for? It seems not linked to zephyr.

**Review by ZhiyuanTang17 — COMMENTED** [review]
I have two problems, please check it out and see if it needs to change code.

File: bee/drivers/adc/src/rtl_common/CMakeLists.txt (hunk 2 above)
Comment (YuzhuoLiuRTK) [reply bee/drivers/adc/src/rtl_common/CMakeLists.txt]: Originally, there was a ADC and a HRADC, so ADC_ADC and ADC_HRADC were used to differentiate them. However, HRADC has not been adapted yet, so it has been changed to only retain CONFIG_REALTEK_ADC. This modification has been made.

File: bee/drivers/adc/Kconfig (hunk 1 above)
Comment (YuzhuoLiuRTK) [reply bee/drivers/adc/Kconfig]: Originally, there was a ADC and a HRADC, so ADC_ADC and ADC_HRADC were used to differentiate them. However, HRADC has not been adapted yet, so it has been changed to only retain CONFIG_REALTEK_ADC. This modification has been made.

File: bee/drivers/driver_lib/lib/rtl87x2g/gcc/librtl87x2g_io.a
Comment (YuzhuoLiuRTK) [reply bee/drivers/driver_lib/lib/rtl87x2g/gcc/librtl87x2g_io.a:1]: This library is used in the MDK project to store driver-related functions, which may not be needed in the GCC project. Therefore, I will delete it.

**Review by ZhiyuanTang17 — APPROVED** [review]
approve

**Review by RSGraceGuan — APPROVED** [review]

**Pull request merged**